lib/mongoid/matchable/default.rb

```diff
@@ -69,6 +69,32 @@ def determine(value, operator)
           attr ? attr.send(operator, first(value)) : false
         }
       end
+
+      # Convenience method for checking _matches? on a Document or a Hash.
+      #
+      # @example Does the document of type Document or Hash match?
+      #   safe_matches?(default, "a", 1)
+      #   safe_matches?(:a => 1, "b", 2)
+      #
+      # @param [ Document, Hash ] document The object of type Document or Hash to call _matches? on
+      # @param [ String ] key The key
+      # @param [ Object ] value The value to check if it matches.
+      #
+      # @return [ true, false ] True if matches, false if not.
+      #
+      # @since 7.0.4
+      def safe_matches?(document, key, value)
```

Comment:
why is this called `safe_matches?`. What is safe about it (or unsafe about the old way.) Maybe there's a better name like `recursive_matches`

Reply:
I actually couldn't land on a good name for the method. (Safe bc it doesn't raise errors on hashes?). I can rename to recursive_matches since that seems better